libs/vkd3d/command.c

```diff
@@ -2955,6 +2956,114 @@ static HRESULT STDMETHODCALLTYPE d3d12_command_list_Close(d3d12_command_list_ifa
     return S_OK;
 }
 
+static bool d3d12_command_list_find_query(struct d3d12_command_list *list,
+        VkQueryPool vk_pool, uint32_t index, size_t *out_pos)
+{
+    const struct vkd3d_query_range *range;
```

Comment:
What is the motivation to roll a custom binary search here? The fallback VA allocator uses bsearch(). Although, since this is kinda perf critical, I guess we can justify not using C stdlib (which cannot inline the comparator).

Reply:
If the query in question is not included in any range, bsearch would return `NULL`, but we need to know where to insert a new range in that case.

Reply:
Fair enough.